coriolis/tests/minion_manager/rpc/test_server.py

```diff
@@ -0,0 +1,201 @@
+
+
+@ddt.ddt
+class MinionManagerServerEndpointTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Minion Manager RPC server."""
+
+    def setUp(self, *_, **__):
+        super(MinionManagerServerEndpointTestCase, self).setUp()
+        self.server = server.MinionManagerServerEndpoint()
+
+    @mock.patch.object(db_api, "get_minion_pools")
+    @ddt.file_data(
+        "data/validate_minion_pool_selections_for_action_config.yaml"
+    )
+    @ddt.unpack
+    def test_validate_minion_pool_selections_for_action(
+            self,
+            mock_get_minion_pools,
+            config,
+            expected_exception,
+    ):
+        action = config.get("action")
+        minion_pools = config.get("minion_pools", [])
+
+        mock_get_minion_pools.return_value = [
+            mock.MagicMock(
+                **pool,
+            ) for pool in minion_pools
+        ]
+
+        if expected_exception:
+            exception_type = getattr(exception, expected_exception)
+            self.assertRaises(
+                exception_type,
+                self.server.validate_minion_pool_selections_for_action,
+                mock.sentinel.context,
+                action,
+            )
+            return
+
+        self.server.validate_minion_pool_selections_for_action(
+            mock.sentinel.context,
+            action,
+        )
+
+    @mock.patch.object(uuid, "uuid4", return_value="new_machine")
+    @mock.patch.object(
+        server.MinionManagerServerEndpoint,
+        "_add_minion_pool_event")
+    @mock.patch.object(db_api, "add_minion_machine")
+    @mock.patch.object(db_api, "set_minion_machines_allocation_statuses")
+    @ddt.file_data(
+        "data/make_minion_machine_allocation_subflow_for_action.yaml"
+    )
+    @ddt.unpack
+    def test_make_minion_machine_allocation_subflow_for_action(
+            self,
+            mock_set_minion_machines_allocation_statuses,
+            mock_add_minion_machine,
+            mock_add_minion_pool_event,
+            mock_uuid4,
+            config,
+            expect
+    ):
+        expected_exception = expect.get("exception")
+        expected_result = expect.get("result")
+
+        minion_pool = testutils.DictToObject(config.get("minion_pool"))
+        action_instances = config.get("action_instances")
+
+        args = [
+            mock.sentinel.context,
+            minion_pool,
+            mock.sentinel.action_id,
+            action_instances,
+            mock.sentinel.subflow_name,
+        ]
+
+        if expected_exception:
+            exception_type = getattr(exception, expected_exception)
+            self.assertRaises(
+                exception_type,
+                self.server._make_minion_machine_allocation_subflow_for_action,
+                *args)
+            return
+
+        result = self.server\
+            ._make_minion_machine_allocation_subflow_for_action(
+                *args)
+
```

Comment:
Also assert `mock_add_minion_pool_event` gets called. In case it's too complicated to also check these calls, feel free to refactor the code.

Reply:
`delete_minion_machine` call is also not checked

Reply:
The tested method `_make_minion_machine_allocation_subflow_for_action` clearly needs some kind of refactoring, as you're missing some Exception raising case coverage

Reply:
> `delete_minion_machine` call is also not checked

I need to create a separate test to handle this, since an exception must be thrown, but I saw that here:

https://github.com/cloudbase/coriolis/blob/master/coriolis/minion_manager/rpc/server.py#L706

`new_machine_db_entries_added` is used as a list of strings, while here:

https://github.com/cloudbase/coriolis/blob/master/coriolis/minion_manager/rpc/server.py#L718 and here:
https://github.com/cloudbase/coriolis/blob/master/coriolis/minion_manager/rpc/server.py#L742

is used as a list of the machine object, thus my test fails with this error:
AttributeError: 'str' object has no attribute 'id'